crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -585,14 +615,39 @@ impl From<&Indices> for IndexFormat {
     }
 }
 
+bitflags::bitflags! {
+    #[repr(transparent)]
+    pub struct VertexLayoutSpriteKey: u32 {
+        const NONE              = 0;
+        const COLORED           = (1 << 0);
+    }
+}
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum VertexLayoutKey {
+    Mesh(u64),
+    Sprite(VertexLayoutSpriteKey),
+    Ui,
```

Comment:
I don't think we should hard code layouts like this. The UI vertex layout shouldn't be any more "special" than the layout defined in a 3rd party crate.

Reply:
Do we want to store the attribute layouts on each individual sprite? On the sprite-batch? Somewhere else? Similar questions for UI.

Reply:
I think UI and sprite layouts should be hard coded for now in the specialize function / not a part of the key:
* different layouts would need different buffers alongside different batches. this would require more restructuring than im comfortable with at this point in the game.
* when drawing hundreds of thousands of sprites, per-sprite CPU usage becomes a real problem. adding more work to the key hash would probably register on benchmarks